### ISSUE TITLE
Reduce number of VM_classHasBeenReplaced messages

### ISSUE THREAD

**Issue by mpirvu** [eclipse/openj9#5650]
`VM_classHasBeenReplaced` messages are issued by the implementation of `TR_J9ServerVM::classHasBeenReplaced(TR_OpaqueClassBlock *clazz)`.
The implementation of `TR_J9VM::classHasBeenReplaced` is
```
TR_J9VM::classHasBeenReplaced(TR_OpaqueClassBlock * clazzPointer)
   {
   return (J9CLASS_FLAGS(TR::Compiler->cls.convertClassOffsetToClassPtr(clazzPointer)) & J9AccClassHotSwappedOut) != 0;
   }
```
We already cache the value of `clazz->classDepthAndFlags` and therefore we have the value of `J9AccClassHotSwappedOut` flag. The question is whether this flag can change. I would argue that when this flag changes the HCR hook is called on the client which will send a message with the classes that have been replaced and invalidate the corresponding entry at the server. Moreover, any HCR event squashes compilations in progress.

**Comment by dmitry-ten** [comment]
Possibly a beginner issue?

**Comment by dmitry-ten** [comment]
I looked at all of the places where `J9AccClassHotSwappedOut` is set, and they all are related to HCR, so I think Marius is right that we should be able to safely use the cached result.

I also ran a bunch of sanity tests that use HCR and jvmti, and noticed that in all of these tests, the `classHasBeenReplaced` always returns false. This seems to confirm that setting the flag always results in compilation being terminated, but then it's not clear why this query is even needed.
